install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -11,144 +11,396 @@
-----
-====
-
-All containers in the above pod will be members of the group *1234* and will
-have access to the volume regardless of what user the container runs as. This is
-useful for volumes which do not allow you to change the export's ownership on
-the client, such as NFS and GlusterFS.
-
-== FSGroup
-
-`*FSGroup*` stands for file system group. It is a supplemental group as defined
-above, but with some extra functionality. If `*FSGroup*` is specified, it will
-have the following effects on volumes which support ownership management:
-
-* The owning group of the volume will be set to the specified `*FSGroup*`.
-* Newly created files in the volume will also be owned by `*FSGroup*`.
-* Read and write permissions will be given to `*FSGroup*`.
-* The specified `*FSGroup*` will be added to the pod's list of supplemental
-groups.
-
-This gives your pod, and any other pods with the same `*FSGroup*`, access to the
-volume.
-
-[IMPORTANT]
-====
-Again, it is recommended to allow OpenShift to automatically allocate an
-`*FSGroup*` unless you have a specific use case which requires you to override
-it in the pod definition.
-====
-
-`*FSGroup*` can be specified as follows:
-
-====
-[source,yaml]
-----
-apiVersion: v1
-kind: Pod
+  - name: nfs <1>
+    nfs:
+      server: <nfs-server-ip-or-host>
+      path: /opt/nfs <5>
+```
+<1> name of the volume mount, must match the name in the `volumes` section.
+<2> nfs export path as seen in the container.
+<3> pod global security context: applies to all containers in pod. Note: each container can also define its
+`securityContext`; however, group ids are global to the pod, and cannot be defined for individual containers.
+<4> supplemental groups, which is an array of ids, is set to 5555. This grants group access to the export.
+<5> actual nfs export path on the nfs server.
+
+All containers in the above pod (assuming the matching SCC or project allows the group *5555*) will be
+members of the group *5555*, and will have access to the volume, regardless of the container's user id.
+However, the assumption above is critical. Sometimes, the SCC does not define a range of allowable group
+ids but requires group id validation (due to `supplementalGroups` set to _MustRunAs_; note this is
+not the case for the _restricted_ SCC). And, the namespace will not likely allow a group id of 5555
+(unless the project has been customized for access to this NFS export). So, in this scenario, the above
+pod will fail because its group id of *5555* is not within the SCC's or the namespace's range of allowed
+group ids. 
+
+[[scc-supplemental-groups]]
+==== Supplemental Groups and Custom SCCs
+To remedy this situation a custom SCC can be created such that a min and max group id are defined,
+id range checking is enforced, and the group id of 5555 is allowed. It is considered a better
+practice to create new SCCs versus modifying a predefined SCC, or changing the range of allowed
+ids in the predefined projects. 
+
```

Comment:
Is it worth mentioning what the user gets with the defaults out of the box? Also that supp groups are meant to accommodate existing permissions for shared storage...?